Formula/clazy.rb

```diff
@@ -0,0 +1,69 @@
+class Clazy < Formula
+  desc "Qt oriented static code analyzer"
+  homepage "https://www.kdab.com/"
+  url "https://download.kde.org/stable/clazy/1.9/src/clazy-1.9.tar.xz"
+  sha256 "4c6c2e473e6aa011cc5fab120ebcffec3fc11a9cc677e21ad8c3ea676eb076f8"
+  license "LGPL-2.0-or-later"
+  head "https://invent.kde.org/sdk/clazy.git"
+
+  livecheck do
+    url :head
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
+  depends_on "cmake"   => [:build, :test]
+  depends_on "llvm@11" => [:build, :test]
+  depends_on "qt"      => :test
+
+  depends_on "coreutils"
+
+  uses_from_macos "libxml2"
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
+
+  def install
+    # avoid use llvm libc++
+    ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm@11"].opt_lib
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"CMakeLists.txt").write <<~EOS
+      cmake_minimum_required(VERSION #{Formula["cmake"].version})
+
+      project(test VERSION 1.0.0 LANGUAGES CXX)
+
+      set(CMAKE_CXX_STANDARD 17)
+      set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+      set(CMAKE_AUTOMOC ON)
+      set(CMAKE_AUTORCC ON)
+      set(CMAKE_AUTOUIC ON)
+
+      find_package(Qt6 COMPONENTS Core REQUIRED)
+
+      add_executable(test
+          test.cpp
+      )
+
+      target_link_libraries(test PRIVATE Qt6::Core
+      )
+    EOS
+
+    (testpath/"test.cpp").write <<~EOS
+      #include <QtCore/QString>
+      void test()
+      {
+          qgetenv("Foo").isEmpty();
+      }
+      int main() { return 0; }
+    EOS
+
+    ENV["CLANGXX"] = Formula["llvm@11"].opt_bin/"clang++"
+    system "cmake", "-DCMAKE_CXX_COMPILER=#{bin}/clazy", "."
```

Comment:
Is this formula usable if I don't have `llvm@11` installed?

Reply:
Clang plugins doesn't  linked against `libclang-cpp` directly. Apple clang could load it if the version matched.

Reply:
Shouldn't Big Sur `clang` have a matching API version?

Reply:
Xcode clang cannot load this plugin.